pkg/combustion/templates/14-embedded-registry.sh.tpl

```diff
@@ -18,11 +20,67 @@ cat <<- EOF > /etc/systemd/system/eib-embedded-registry.service
   User=root
   WorkingDirectory=/opt/hauler
   ExecStartPre=/usr/local/bin/hauler store load {{ .EmbeddedRegistryTar }}
-  ExecStart=/usr/local/bin/hauler store serve -p {{ .Port }}
+  ExecStart=/usr/local/bin/hauler store serve registry -p {{ .RegistryPort }}
   Restart=on-failure
 
   [Install]
   WantedBy=multi-user.target
 EOF
+systemctl enable eib-embedded-registry.service
+
+# This serves the hauler registry artifacts as a file server
+{{- if .ChartsDir }}
+cat <<- EOF > /etc/systemd/system/eib-embedded-fileserver.service
+  [Unit]
+  Description=Load and Serve Embedded File Server
+  After=network.target
+
+  [Service]
+  Type=simple
+  User=root
+  WorkingDirectory=/opt/hauler
+  ExecStartPre=/bin/sleep 30
+  ExecStart=/usr/local/bin/hauler store serve fileserver -p {{ .FileServerPort }}
```

Comment:
Another implementation idea, what about running this inside Kubernetes? This way you wouldn't have to look for the host IP at all.